packages/shared/src/utils/contract/contract.ts

```diff
@@ -118,6 +118,14 @@ export const getCurrentTick = (contract_info: TContractInfo) => {
     return !current_tick || current_tick < 0 ? 0 : current_tick;
 };
 
+export const getLastContractMarkerIndex = (markers: TContractStore[] = []) => {
+    const sorted_markers = [...markers].sort(
+        (a, b) => Number(b.contract_info.date_start) - Number(a.contract_info.date_start)
+    );
+    const index = sorted_markers[0].contract_info.date_start ? markers.indexOf(sorted_markers[0]) : -1;
+    return index >= 0 ? index : markers.length - 1;
```

Comment:
`return sorted_markers[0].contract_info.date_start ? markers.indexOf(sorted_markers[0]) :  markers.length - 1;`
Not sure if it's good idea

Reply:
@kate-deriv unfortunately, it won't work because if I apply it and `markers.indexOf(sorted_markers[0])` returns `-1`, the returned result will be `-1` while I need to return `markers.length - 1` in this case

Reply:
Sorry, I thought` markers.indexOf(sorted_markers[0])` will always return something (not `-1`) as it's a copy of `markers` array (but sorted)

Reply:
Okay, then LGTM